packages/block-editor/src/components/block-list/index.js

```diff
@@ -248,13 +249,24 @@ class BlockList extends Component {
 					const isBlockInSelection = hasMultiSelection ?
 						multiSelectedBlockClientIds.includes( clientId ) :
 						selectedBlockClientId === clientId;
+					const isFirstMultiSelected = multiSelectedBlockClientIds[ 0 ] === clientId;
+					const shouldShowInsertionPoint = ! isMultiSelecting && (
+						( isBlockInSelection && isFirstMultiSelected ) ||
+						! isBlockInSelection
+					);
 
 					return (
 						<BlockAsyncModeProvider
 							key={ 'block-' + clientId }
 							clientId={ clientId }
 							isBlockInSelection={ isBlockInSelection }
 						>
+							{ shouldShowInsertionPoint && (
+								<BlockInsertionPoint
+									clientId={ clientId }
+									rootClientId={ rootClientId }
+								/>
+							) }
```

Comment:
i'm not sure I understand the reasoning? It's clearer for me to have one unique component to show the Block UI, that component can be split into smaller components if needed but while looping through blocks I prefer to have "one entry point" that needs to be optimized.

Also, the BlockListBlock is a pure component which means it doesn't rerender when the list block does, this change wille force us to think about this for the BlockInsertionPoint as well.

Can you clarify a bit more about the reasoning and what is this trying to solve?

Reply:
Currently the "in-between" inserter is nested inside the next block rather than literally in between blocks. This has some problems:

* It prevents us from making the block DOM a lot lighter and moving towards a thin wrapper without any controls in it. (See controls in popover PR.)
* The tab order is a bit strange. First you select the next block, then you select the inserter. In this PR you first tab to the insert, then you select the next block.
* From a data structure point of view, it makes more sense to have "in-between" inserters in between blocks rather than being part of a block. The block component shouldn't manage this inserter component.

Reply:
From a styling point of view, this also makes the inserter appear exactly in the middle of two blocks, no matter what the margins are.

Reply:
That could work too if you prefer.